### ISSUE TITLE
Set global variable does not work at the folder level

### ISSUE THREAD

**Issue by sevencardz** [a85/POSTMan-Chrome-Extension#4909]
Setting a global variable is not working at the folder level.

**Steps to Reproduce:**
Create a folder within a collection and give it the following test, then run it:
```
pm.test('Set global variable. Initial value: ' + pm.variables.get('testGlobalVar'), function() {
	pm.globals.set('testGlobalVar', 'SUCCESS');
	pm.expect(pm.variables.get('testGlobalVar')).to.eql('SUCCESS');
	console.log('globalvar: ' + pm.variables.get('testGlobalVar'));
});
```
The global variable should be set to 'SUCCESS', but on subsequent tests the initial value remains unchanged. Checking the value from within the Manage Environments pane also shows the value is unchanged. This is despite the value passing the equality test on the next line and printing out 'SUCCESS' to the console.

Moving the test into the Test tab for a test *within* a folder exhibits the desired behavior - the global value is set to 'SUCCESS' as expected.

**App information (please complete the following information):**
 - App Type: Native App
 - Postman Version 6.2.2
 - OS: Windows 10 Pro

**Additional context**
I have 'Automatically persist variable values' turned on in Settings.

**Comment by harryi3t** [comment]
@sevencardz Thanks for reporting the issue.
I tried reproducing on my end but was unable to do so.
Here's what I have tried:

### I am setting a global var from a folder pre-request script
![image](https://user-images.githubusercontent.com/5207331/43353034-afcd353a-924b-11e8-89e7-f7d02f318721.png)

### I am setting a global var from a folder test script
![image](https://user-images.githubusercontent.com/5207331/43353037-b5241b34-924b-11e8-8cec-cff9c3e38d68.png)

### After running the request from the builder
![image](https://user-images.githubusercontent.com/5207331/43353042-d3857b4a-924b-11e8-8048-63e533f79604.png)

What am I missing here? Could you share the screenshots or even better if you could share the collection?

**Comment by vr00** [comment]
Same/similar problem here. I open saved request (2 folders deep in a Collection), click Send. Then in response body which is JSON (pretty) I select one string value (JWT access token) > right click > Set {Environment Name} > {Variable}.

Expected result:
Variable is saved

Actual result:
Nothing happens

Postman version 6.2.2, didn't notice that behavior before. Also, the sync icon flashes at that moment, but if you go to Manage Environments > {Environment Name} you can see the variable hasn't changed.

**Comment by sevencardz** [comment]
@harryi3t Thank you for your help with this. Now things get really interesting. :) 

I have included two functionally identical collections. Collection 1 exhibits the _undesired_ behavior (the global variable value does NOT persist). However, collection 2 works _correctly_ and the value does indeed persist. The only difference between these two collections is that I did not add any test scripts to the Request in collection 2. I added test scripts to the Request in collection 1, but I then removed them (with Postman UI) - to see if the issue persists after that modification and indeed it does. Upon comparing the resulting exported JSON collection files, I see that collection 1 has 'event' objects defined, whereas collection 2 does not, among other differences. These differences seem to have some significance as to whether the bug will manifest or not.
[Test Global Variable Postman Collections.zip](https://github.com/postmanlabs/postman-app-support/files/2238468/Test.Global.Variable.Postman.Collections.zip)

**Comment by harryi3t** [comment]
@sevencardz @vr00 Thank you, using your feedback I was able to pinpoint the issue.
This bug affects the collections having scripts at multiple levels (collection, folder & request level).
So if you have a collection which has both folder scripts and request scripts, only the modifications made by request script are persisted.
If a collection only has folder scripts or only collection scripts then there will no issue.

We are working on the fix right now. Thanks again for your inputs.

**Comment by harryi3t** [comment]
@sevencardz @vr00 We have released the fix in our canary build which you can download from https://www.getpostman.com/canary

Please could you verify the fix?
Also, the same will be available in the next stable version of Postman

**Comment by a85** [comment]
@harryi3t Can we write a short post-mortem here about the bug i.e. why this happened?

**Comment by vr00** [comment]
@harryi3t Tested canary release, works as expected. Good job on that one 👍 

Thanks

**Comment by sevencardz** [comment]
@harryi3t I have also tested the canary release and all is working as expected now. Thank you!

**Comment by harryi3t** [comment]
@a85 Sure, here's a short post-mortem about the bug

#### Q. What was the bug?
Changes made to variables (environment or globals) were not persisting for users having scripts at multiple levels (collection, folder, and request levels)

#### Q. Why did it happen?
Postman is built on top of [postman-runtime](https://github.com/postmanlabs/postman-runtime) which is its execution engine. In earlier versions after a request was executed, Postman was consuming the complete snapshot of variables. In the newer version of the runtime, for performance reasons, we now send the variable diffs (only the modified variables). There was a bug in the app code where we were not collating all the diffs but only taking the last one, which resulted in any modifications made in the folder/collection level scripts to be discarded.

#### Q. How are you going to prevent this from happening in the future?
We have the unit and end-to-end integration tests. We have added this case in our tests so that it would not regress in the future.

**Comment by harryi3t** [comment]
@sevencardz @vr00 This is fixed in the latest version of Postman 6.2.3
